src/common/screen.rs

```diff
@@ -303,4 +310,9 @@ impl Screen {
             }
         }
     }
+    pub fn change_input_mode(&mut self, input_mode: InputMode) {
+        self.input_mode = input_mode;
+        self.get_active_tab_mut().unwrap().input_mode = self.input_mode;
+        self.render();
```

Comment:
In the input handler where we send the `ChangeInputMode` instruction, we already explicitly send a `Render` instruction later... maybe we can remove this extra render then?